server/http/interceptors/interceptors.go

```diff
@@ -106,6 +106,18 @@ func Authenticate(env environment.Env, next http.Handler) http.Handler {
 	})
 }
 
+func SetAPIKeyFromAuthHeader(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		authHeader := r.Header.Get("Authorization")
+		// cannot use auth.APIKeyHeader because of circular dependency
+		apiKey := strings.TrimPrefix(authHeader, "x-buildbuddy-api-key ")
+		if apiKey != "" && r.Header.Get("x-buildbuddy-api-key") == "" {
+			r.Header.Set("x-buildbuddy-api-key", apiKey)
+		}
+		next.ServeHTTP(w, r)
+	})
+}
+
```

Comment:
maybe a cleaner way to do this would be to accept an API key set as a bearer token in AuthenticatedHTTPContext -- so if someone made an HTTP request to use with:

`header: Authorization <api_key_key>`

We'd do the right thing.

Reply:
Done. I added the support in AuthenticatedHTTPContext to support

`header: Authorization: x-buildbuddy-api-key <api_key>`

I chose to use x-buildbuddy-api-key, so that in the future if we ever want to use bearer token for HTTP we won't have a problem